File: home/v/.config/nvim/after/plugin/normon.lua
local normon = require("normon")
-- cgn on current word/selection
normon("<leader>n", "cgn")
normon("<leader>N", "cgN")

-- macro on current word/selection
normon("<leader>q", "qq")
normon("<leader>Q", "qq", { backward = true })

-- improved * and #
-- escapes special characters and does not ignore case
normon("*", "n", { clearSearch = true })
normon("#", "n", { backward = true, clearSearch = true })

-- Jake has this:
--return {
--    "jake-stewart/normon.nvim",
--    keys = {
--        {"<leader>n", mode = {"n", "v"}},
--        {"<leader>N", mode = {"n", "v"}},
--        {"<leader>q", mode = {"n", "v"}},
--        {"<leader>Q", mdoe = {"n", "v"}},
--        {"*", mode = {"n", "v"}},
--        {"#", mode = {"n", "v"}},
--        {"<leader>dgn", mode = {"n", "v"}},
--        {"<leader>dgN", mdoe = {"n", "v"}},
--    },
--    config = function()
--        local normon = require("normon")
--        normon("<leader>n", "cgn")
--        normon("<leader>N", "cgN")
--        normon("<leader>q", "q")
--        normon("<leader>Q", "q", {backward = true})
--        normon("*", "n", {clearSearch = true})
--        normon("#", "n", {backward = true, clearSearch = true})
--
--        normon("<leader>dgn", "dgn")
--        normon("<leader>dgN", "dgN")
--    end
--}

File: home/v/.config/nvim/typescript/src/plugins/normon.ts
export const normon = {
    1: "jake-stewart/normon.nvim",
    dir: "~/clones/normon.nvim",
    keys: [
        { 1: "<leader>n", mode: ["n", "v"] },
        { 1: "<leader>N", mode: ["n", "v"] },
        { 1: "<leader>q", mode: ["n", "v"] },
        { 1: "<leader>Q", mode: ["n", "v"] },
        { 1: "*", mode: ["n", "v"] },
        { 1: "#", mode: ["n", "v"] },
    ],
    config: () => {
        const normon = lua.require("normon")
        normon("<leader>n", "cgn")
        normon("<leader>N", "cgN")
        normon("<leader>q", "q")
        normon("<leader>Q", "q", { backward: true })
        normon("*", "n")
        normon("#", "n", { backward: true })

        vim.keymap.set("v", "g*", "*")
        vim.keymap.set("v", "g#", "#")
    }
}
